### PR TITLE
[iOS][font] Fix some vector icons using incorrect fonts

### LINKED ISSUE #28693: [ios] FontAwesome5 does not import properly with SDK 51
### Minimal reproducible example

https://github.com/anthonyn60/font-repro

### What platform(s) does this occur on?

iOS

### Did you reproduce this issue in a development build?

Yes

### Summary

Fonts and icons don't load when imported from `FontAwesome5`. The font falls back to the default, while the icon will show a square with a question mark in it.

Screenshot 2024-05-08 at 4 40 49 PM

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.4.1
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 21.6.1 - ~/.nvm/versions/node/v21.6.1/bin/node
      Yarn: 1.22.21 - ~/.nvm/versions/node/v21.6.1/bin/yarn
      npm: 10.6.0 - ~/.nvm/versions/node/v21.6.1/bin/npm
      Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/opt/rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.4, iOS 17.4, macOS 14.4, tvOS 17.4, visionOS 1.1, watchOS 10.4
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10671973
      Xcode: 15.3/15E204a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.0 => 51.0.0
      expo-router: ~3.5.10 => 3.5.10
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.74.1 => 0.74.1
      react-native-web: ~0.19.10 => 0.19.11
    npmGlobalPackages:
      eas-cli: 7.8.2
    Expo Workflow: managed

### Expo Doctor Diagnostics

Anthonys-MacBook-Pro-2:font-repro anthonynguyen$ npx expo-doctor@latest

> font-repro@1.0.0 npx
> expo-doctor

(node:93133) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check dependencies for packages that should not be installed directly
✔ Check for issues with metro config
✔ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK
✔ Check native tooling versions

Didn't find any issues with the project!

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Fixes #28693 

# How

When the swizzled `UIFont.fontNames(forFamilyName:)` function cannot find any font names for the aliased font family, we can assume it was not a family name but a font name in which case we can return an array with this font name instead of an empty array. Thanks to this, the font resolution in RN will not enter this if: https://github.com/facebook/react-native/blob/00b251b8ee8752b82a536faabc424d92f7034ebe/packages/react-native/React/Views/RCTFont.mm#L448 but will enter this one: https://github.com/facebook/react-native/blob/00b251b8ee8752b82a536faabc424d92f7034ebe/packages/react-native/React/Views/RCTFont.mm#L471 which is the expected behavior as it would bypass the unwanted logic.
In other words, this change shifts the responsibility of handling font names passed as family names to us.

I also reverted #28407 which is no longer necessary and added some FontAwesome icons to our expo-font examples in native-component-list app.

# Test Plan

Tested repros from #28693, #28407 and the new examples on both old and new architectures

Simulator Screenshot - iPhone 15 Pro Max - 2024-05-10 at 15 40 19

Also thanks for @alanjhughes for helping me investigate this issue and test various cases and fonts.

### CODE REVIEW

**Review by alanjhughes — APPROVED** [review]

**Review by brentvatne — APPROVED** [review]

File: apps/native-component-list/src/screens/FontScreen.tsx [hunk 1]
```diff
@@ -1,25 +1,10 @@
...
-        <Ionicons name="logo-facebook" size={25} />
-        <Ionicons name="logo-apple" size={25} />
-        <Ionicons name="logo-amazon" size={25} />
-        <Ionicons name="logo-npm" size={25} />
-        <Ionicons name="logo-google" size={25} />
-        <Ionicons name="alarm" size={25} />
-      </View>
-
       <View style={{ paddingVertical: 10, paddingHorizontal: 15, flex: 1 }}>
         <Text style={{ fontFamily: 'space-mono', fontSize: 16 }}>
```
Comment (brentvatne) [inline apps/native-component-list/src/screens/FontScreen.tsx:9]: we might want to also add a font that we name something totally unrelated like "A-Custom-Font-Name" to verify that we don't regress in that functionality

**Comment by Chayma-Soussi** [comment]
hey please i have the same problem when i update sdk 51 this is my package.json  {
  "name": "lincare-app",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web",
    "email": "email dev"
  },
  "dependencies": {
    "@expo/vector-icons": "^14.0.2",
    "@react-email/components": "0.0.17",
    "@react-native-async-storage/async-storage": "^1.23.1",
    "@react-native-community/blur": "^4.4.0",
    "@react-navigation/bottom-tabs": "^6.5.20",
    "@react-navigation/native": "^6.1.16",
    "@react-navigation/native-stack": "^6.9.26",
    "@react-navigation/stack": "^6.3.28",
    "@types/moment": "^2.13.0",
    "@types/react": "~18.2.45",
    "axios": "^1.6.8",
    "expo": "~51.0.2",
    "expo-auth-session": "~5.5.2",
    "expo-constants": "~16.0.1",
    "expo-crypto": "~13.0.2",
    "expo-dev-client": "~4.0.11",
    "expo-font": "~12.0.4",
    "expo-image-picker": "~15.0.4",
    "expo-linear-gradient": "~13.0.2",
    "expo-mail-composer": "~13.0.1",
    "expo-notifications": "~0.28.1",
    "expo-random": "~14.0.1",
    "expo-status-bar": "~1.12.1",
    "expo-updates": "~0.25.9",
    "expo-web-browser": "~13.0.3",
    "formik": "^2.4.5",
    "moment": "^2.30.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-email": "2.1.2",
    "react-native": "^0.73.0",
    "react-native-animatable": "^1.4.0",
    "react-native-calendars": "^1.1304.1",
    "react-native-gesture-handler": "~2.16.1",
    "react-native-modal-datetime-picker": "^17.1.0",
    "react-native-reanimated": "3.10.0",
    "react-native-safe-area-context": "4.10.1",
    "react-native-screens": "3.31.1",
    "react-native-snap-carousel": "^3.9.1",
    "react-native-svg": "15.2.0-rc.0",
    "react-native-vector-icons": "^10.1.0",
    "react-native-web": "~0.19.6",
    "styled-components": "^6.1.8",
    "typescript": "~5.3.3",
    "undefined": "@react-native-async-storage/async-storage/async-storage",
    "victory-native": "^40.1.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@types/react": "~18.2.45",
    "@types/react-native": "^0.73.0"
  },
  "private": true
}

**Comment by Chayma-Soussi** [comment]
in android
